numba/targets/arrayobj.py

```diff
@@ -188,7 +187,9 @@ def getiter_array(context, builder, sig, args):
     iterobj.index = indexptr
     iterobj.array = array
 
-    return iterobj._getvalue()
+    res = iterobj._getvalue()
+    out = impl_ret_borrowed(context, builder, sig.return_type, res)
```

Comment:
Why is it borrowed? Is it the array object which is borrowed? As far as I can see the iterator object itself is new.